code/modules/admin/holder2.dm

```diff
@@ -95,27 +104,43 @@ GLOBAL_PROTECT(href_token)
 		disassociate()
 		add_verb(C, /client/proc/readmin)
 
-/datum/admins/proc/associate(client/C)
+/datum/admins/proc/associate(client/client)
 	if(IsAdminAdvancedProcCall())
 		var/msg = " has tried to elevate permissions!"
 		message_admins("[key_name_admin(usr)][msg]")
 		log_admin("[key_name(usr)][msg]")
 		return
 
-	if(istype(C))
-		if(C.ckey != target)
-			var/msg = " has attempted to associate with [target]'s admin datum"
-			message_admins("[key_name_admin(C)][msg]")
-			log_admin("[key_name(C)][msg]")
-			return
-		if (deadmined)
-			activate()
-		owner = C
-		owner.holder = src
-		owner.add_admin_verbs() //TODO <--- todo what? the proc clearly exists and works since its the backbone to our entire admin system
-		remove_verb(owner, /client/proc/readmin)
-		owner.init_verbs() //re-initialize the verb list
-		GLOB.admins |= C
+	if(!istype(client))
+		return
+
+	if(client?.ckey != target)
+		var/msg = " has attempted to associate with [target]'s admin datum"
+		message_admins("[key_name_admin(client)][msg]")
+		log_admin("[key_name(client)][msg]")
+		return
+
+	var/result_2fa = check_2fa(client)
+	if (!result_2fa[RESULT_2FA_VALID])
+		blocked_by_2fa = TRUE
+		alert_2fa_necessary(client)
+		start_2fa_process(client, result_2fa[RESULT_2FA_ID])
+
+		return
```

Comment:
By doing this here, you cause admins to get the player rank in the database, and this breaks tgdb because it doesn't look at the admin database, but the last admin rank column of the player table. This is done this way so acs would get access to tgdb while admin'ed up by an admin trainer.

Reply:
simulating a call to `sync_lastadminrank()` on verify will solve this.

Reply:
I'm a bit confused on the details, but I can add that proc call on verify, sure.

I'm working on this again tomorrow.